Review

maksim533 opened the pull request.

maksim533 committed:
проект перед ревью

maksim533 committed:
fix: после ревью

maksim533 committed:
Тесты к 12 проектной работе

gennady-bars commented on src/services/slices/constructorSlice.test.ts:
```diff
@@ -0,0 +1,162 @@
+      },
+      isLoading: false,
+      orderRequest: false,
+      orderModalData: null,
```
`initialState ` не нужно создавать в тестах опять.  Нужно импортировать `initialState` из файла редьюсера. Там уже создан начальный стейт для каждого редьюсера

gennady-bars commented on src/services/slices/ingredientsSlices.ts:
```diff
@@ -0,0 +1,50 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { TIngredient } from '@utils-types';
+
+export interface ingredientsState {
```
`ingredientsState`

Кастомные типы `typescript` принято называть в `PascalCase` (с **большой буквы**)

gennady-bars commented on cypress/e2e/constructor/constructor.cy.ts:
```diff
@@ -0,0 +1,92 @@
+    cy.intercept('GET', `/api/ingredients`, {
+      fixture: 'ingredients.json'
+    }).as('getIngredients');
+    cy.visit('http://localhost:4000');
```
`http://localhost:4000` обычно повторяется несколько раз в коде тестов. Нужно вынести в константу `testUrl`, чтобы можно было одним движением изменить урл для тестов.
Mожно в документации посмотреть, как настроить быстро `baseUrl` для cypress https://docs.cypress.io/api/commands/visit#Visit-is-automatically-prefixed-with-baseUrl

gennady-bars commented on cypress/e2e/constructor/constructor.cy.ts:
```diff
@@ -0,0 +1,92 @@
+    localStorage.removeItem('refreshToken');
+  });
+  it('Проверка добавление ингредиентов в конструктор', () => {
+    cy.get('[data-test="constructor"]').as('constructor');
```
* Отлично, что знаете про `alias` (переменные с `@`)

gennady-bars commented on cypress/e2e/constructor/constructor.cy.ts:
```diff
@@ -0,0 +1,92 @@
+
+  it('Проверка открытия и закрытия модального окна', () => {
+    cy.get('[data-test="constructor-ingredient"]').first().click();
+    cy.get('[data-test="modal"]').should('be.visible');
```
`[data-test="modal"]` дублируется 7 раз

Если **строка-селектор** повторяется более 2х раз в коде, значит, этот селектор нужно выносить в **константу**, чтобы можно было одним движением изменить его и не искать по всему коду дублирования этого селектора. **Нужно исправить это для всех дублирований в тестах**

**Можно лучше**

* Лучше использовать переменные **alias** в тестах, чтобы 1 раз найти элемент и обращаться к нему. Подробнее тут https://docs.cypress.io/api/commands/as#DOM-element 
* Попробуйте сделать команды `cypress`. Подробнее тут https://on.cypress.io/custom-commands 
* Воспользуйтесь лучшими практиками `cypress`. Подробнее тут https://docs.cypress.io/guides/references/best-practices

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на **гитхабе** все комментарии к коду (нужно прокрутить вниз страницу там)

Работа проделана огромная:


* Все юнит-тесты (41 шт) проходят успешно
* Все cypress-тесты проходят успешно
* Отлично, что знаете про `alias` (переменные с `@`)
* Код тестов аккуратный и понятный


но есть некоторые недочеты:

* `initialState ` не нужно создавать в тестах опять.  Нужно импортировать `initialState` из файла редьюсера. Там уже создан начальный стейт для каждого редьюсера
* Кастомные типы `typescript` принято называть в `PascalCase` (с **большой буквы**)
* `http://localhost:4000` обычно повторяется несколько раз в коде тестов. Нужно вынести в константу `testUrl`, чтобы можно было одним движением изменить урл для тестов.
Mожно в документации посмотреть, как настроить быстро `baseUrl` для cypress https://docs.cypress.io/api/commands/visit#Visit-is-automatically-prefixed-with-baseUrl  
* Если **строка-селектор** повторяется более 2х раз в коде, значит, этот селектор нужно выносить в **константу**, чтобы можно было одним движением изменить его и не искать по всему коду дублирования этого селектора. **Нужно исправить это для всех дублирований в тестах**

**Можно лучше**

* Лучше использовать переменные **alias** в тестах, чтобы 1 раз найти элемент и обращаться к нему. Подробнее тут https://docs.cypress.io/api/commands/as#DOM-element 
* Попробуйте сделать команды `cypress`. Подробнее тут https://on.cypress.io/custom-commands 
* Воспользуйтесь лучшими практиками `cypress`. Подробнее тут https://docs.cypress.io/guides/references/best-practices

maksim533 committed:
Исправленные тесты к 12 проектной работе, после ревью

maksim533 committed:
Исправления

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.

maksim533 committed:
feat: обновить .gitignore для

- Добавить полную поддержку зависимостей (npm, yarn, pnpm)
- Включить исключения для сборок (/build/, /dist/, /out/)
- Добавить переменные окружения (.env файлы)
- Исключить IDE файлы (.idea/, .vscode/, Vim swp файлы)
- Добавить OS файлы (.DS_Store, Thumbs.db и др.)
- Включить поддержку тестирования (coverage, nyc_output)
- Добавить Storybook и Cypress исключения
- Включить кеш директории (.npm, .eslintcache, .cache и др.)
- Добавить TypeScript кеш (*.tsbuildinfo)
- Включить пакетные менеджеры (yarn integrity, pnp)
- Добавить поддержку современных фреймворков

maksim533 committed:
docs: создать полноценную документацию для проекта Stellar Burgers

- Добавить подробное описание проекта с эмодзи и бейджами
- Описать технологический стек с версиями всех зависимостей
- Включить детальную структуру проекта с пояснениями
- Создать пошаговое руководство по установке и настройке
- Добавить описание всех доступных скриптов
- Включить информацию о тестировании (Jest, Cypress, Storybook)
- Описать архитектуру Redux и систему аутентификации
- Добавить рекомендации по разработке и деплою
- Включить ссылки на макет в Figma и чеклист проекта
- Создать разделы для вклада в проект и поддержки
